capstone-express/server.js

```diff
@@ -584,16 +583,47 @@ app.get("/returnToPrevious/:taskId", verifyUser, async (req, res) => {
   }
 });
 
-app.delete("/user/deleteAccount/:userId", verifyUser, async(req,res) => {
+app.delete("/user/deleteAccount/:userId", verifyUser, async (req, res) => {
   const userId = req.params.userId;
-  
-  try{
-    await UserModel.deleteOne({userId: userId})
+
+  try {
+    await UserModel.deleteOne({ userId: userId });
     res.status(200).json("Deleted");
-  }catch(error){
+  } catch (error) {
     console.log(error);
   }
-})
+});
+//Delete task, will delete any subtask and children
+async function deleteTask(taskId) {
+  const taskParent = await TaskModel.findById({ _id: taskId });
+
+  //Iterate through subtasks and delete any children
+  if(taskParent.subtask !== null){
+    for (let i = 0; i < taskParent.subtask.length; i++) {
+      await deleteSubtaskAndChildren(taskParent.subtask[i]._id);
```

Comment:
So this is more of a performance issue.  Since you delete one and then wait to delete the next. 
1.  You can use `Promise.all()` Basically call them and store the promise in a array and then call promise all.
2.  I would have made a array of the subtasks._id or based on some condition and delete them in one go using DB.methods like `deleteMany`. 
https://www.mongodb.com/docs/manual/reference/method/db.collection.deleteMany/
https://www.mongodbtutorial.org/mongodb-crud/mongodb-deletemany/